torch_geometric/explain/metrics.py

```diff
@@ -0,0 +1,69 @@
+from typing import Tuple
+
+import torch
+from torch import Tensor, tensor
+from torchmetrics import AUROC, ROC
+
+from torch_geometric.explain import Explanation
+
+
+def get_groundtruth_metrics(
+        explanation: Explanation, groundtruth: Explanation
+) -> Tuple[Tensor, Tensor, Tensor, Tensor, Tensor]:
+    """Computes different accuracy metrics when groundtruth is available.
+
+    .. note::
+        Currently we perform hard thresholding (where the threshold value
+        is set to 0) on explanation and groundtruth masks to get true
+        positives, true negatives, false positives and false negatives.
+        I.e., all values in explanation masks and ground truth masks which
+        are greater than 0 is set to 1.
+
+    Args:
+        explanation (Explanation): Explanation output generated by an explainer
+        groundtruth (Explanation): Groundtruth for explanation
+
+    Returns:
+        A 5-item tuple containing the values corresponding to the calculated
+        accuracy, recall, precision, f1_score and auc on the explanation
+        against the groundtruth.
+    """
+
+    ex_masks = explanation.masks
+    ex_mask_tensor = torch.cat(
+        list(map(lambda x: x.view(-1), ex_masks.values())))
+    gt_masks = groundtruth.masks
+    gt_mask_tensor = torch.cat(
+        list(map(lambda x: x.view(-1), gt_masks.values())))
+    threshold = 0
+    gt_mask_tensor[gt_mask_tensor > threshold] = 1.0
+    roc = ROC(task="binary")
+    fpr, tpr, thresholds = roc(ex_mask_tensor, gt_mask_tensor)
```

Comment:
You're right, we don't need the roc calculation for auroc. I think we have two options here, We can also return the full ROC curve, or we can remove these two lines and just return the AUROC, I leave the choice up to you.

Reply:
ROC curve gives three values: `fpr`, `tpr`, `thresholds`, whereas all other metrics return a single value. To maintain consistency, I have decided to calculate AUROC and return that. Moreover, with the inclusion of ROC curve, our return value will have nested values, in which case I believe returning a dict makes more sense.